### DISCUSSION TITLE
Add more plugin context subscribtions

### DISCUSSION THREAD

**Discussion by SpeckyYT** [OpenRCT2/OpenRCT2#16480] [Feature requests]
I was making an [AutoSplitter](https://github.com/LiveSplit/LiveSplit.Server) plugin for speedrunning in OpenRCT2.
For having some more useful features for speedrunning, it would be really nice to have the following `context.subscribe`s:
```js
context.subscribe("game.scenario_change", function(){
  startTimer()
})
context.subscribe("script.unload", function(){
  stopTimer()
})
```

**Comment by James103** [comment]
Additional plugin hook ideas (originally posted in #11773 but moved here because feature requests go in Discussions and this discussion already matched what I was proposing):

* Scenario objective check (useful for creating custom objectives)
  * `scenario.check_objective`
* New guest spawns (useful to modify guest stats or deny the event for guest control)
  * Already Implemented as `guest.generation`
* Park rating calculation (to modify the park rating formula)
* Park value calculation (to modify the park value formula)

Code examples (internal names subject to change):
```js
context.subscribe("scenario.check_objective", function(){
  // This runs every time the scenario objective is checked.
})
context.subscribe("guest.generation", function(){
  // This runs every time a new guest spawns. The function is passed the ID of the guest.
})
context.subscribe("park.rating.calculate", function(){
  // This runs every time the park rating is calculated by the game.
  // Modifying the park rating here should overwrite the park rating as calculated by the game.
})
context.subscribe("park.value.calculate", function(){
  // This runs every time the park value is calculated by the game.
  // Modifying the park value here should overwrite the park value as calculated by the game.
})
```
